geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/AlterRegionCommand.java

```diff
@@ -88,61 +96,220 @@ public Result alterRegion(
+    existingAttributes.setRegionTimeToLive(
+        combine(existingAttributes.getRegionTimeToLive(), deltaAttributes.getRegionTimeToLive()));
+
+    if (deltaAttributes.getCacheLoader() != null) {
+      if (deltaAttributes.getCacheLoader().equals(DeclarableType.EMPTY)) {
+        existingAttributes.setCacheLoader(null);
+      } else {
+        existingAttributes.setCacheLoader(deltaAttributes.getCacheLoader());
+      }
     }
-    return result;
+
+    if (deltaAttributes.getCacheWriter() != null) {
+      if (deltaAttributes.getCacheWriter().equals(DeclarableType.EMPTY)) {
+        existingAttributes.setCacheWriter(null);
+      } else {
+        existingAttributes.setCacheWriter(deltaAttributes.getCacheWriter());
+      }
+    }
+
+    if (!deltaAttributes.getCacheListeners().isEmpty()) {
+      existingAttributes.getCacheListeners().clear();
+      // only add the new cache listeners to the list when it's an EMPTY cache listener
+      if (!deltaAttributes.getCacheListeners().get(0).equals(DeclarableType.EMPTY)) {
+        existingAttributes.getCacheListeners().addAll(deltaAttributes.getCacheListeners());
+      }
+    }
+
+    if (deltaAttributes.getGatewaySenderIds() != null) {
+      existingAttributes.setGatewaySenderIds(deltaAttributes.getGatewaySenderIds());
+    }
+
+    if (deltaAttributes.getAsyncEventQueueIds() != null) {
+      existingAttributes.setAsyncEventQueueIds(deltaAttributes.getAsyncEventQueueIds());
+    }
+
+    if (deltaAttributes.isCloningEnabled() != null) {
+      existingAttributes.setCloningEnabled(deltaAttributes.isCloningEnabled());
+    }
+
+    EvictionAttributes evictionAttributes = deltaAttributes.getEvictionAttributes();
+    if (evictionAttributes != null) {
+      // we only set the max in the delta's lruEntryCount in the alter region command
+      String newMax = evictionAttributes.getLruEntryCount().getMaximum();
+      EvictionAttributes existingEviction = existingAttributes.getEvictionAttributes();
+
+      // we only alter the max value if there is an existing evication attributes
+      if (existingEviction != null) {
+        if (existingEviction.getLruEntryCount() != null) {
+          existingEviction.getLruEntryCount().setMaximum(newMax);
+        }
+
+        if (existingEviction.getLruMemorySize() != null) {
+          existingEviction.getLruMemorySize().setMaximum(newMax);
+        }
+      }
+    }
+    return true;
+  }
+
+  ExpirationAttributesType getExpirationAttributes(Integer timeout,
+      ExpirationAction action, ClassName expiry) {
+    if (timeout == null && action == null && expiry == null) {
+      return null;
+    }
+    if (expiry != null) {
+      return new ExpirationAttributesType(timeout, action,
+          expiry.getClassName(), expiry.getInitProperties());
+    } else {
+      return new ExpirationAttributesType(timeout, action, null, null);
+    }
+  }
+
+  // this is a helper class to combine the existing with the delta ExpirationAttributesType
```

Comment:
Comment should reflect that this is a `help method`, not `helper class`

Reply:
fixed